app/lib/kafka.server.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Kafka } from 'gcn-kafka'
+
+import { getEnvOrDie, hostname } from './env.server'
+import type { Circular } from '~/routes/circulars/circulars.lib'
+
+const client_id = getEnvOrDie('KAFKA_CLIENT_ID')
+const client_secret = getEnvOrDie('KAFKA_CLIENT_SECRET')
+const domain = hostname as
+  | 'gcn.nasa.gov'
+  | 'test.gcn.nasa.gov'
+  | 'dev.gcn.nasa.gov'
```

Comment:
This is not quite correct; `hostname` might `'localhost'`. We have logic to determine the appropriate value for `domain` in ClientSampleCode.tsx, but it's written as a hook and can only be used in a React component. Please refactor that and provide both a hook or a non-hook version.